src/settings.py

```diff
@@ -404,48 +214,48 @@ def trigger_post_change_hook(setting_name, old_value, new_value):
 # ===========================================
 
 
-@post_change_hook("ssowat.panel_overlay.enabled")
-@post_change_hook("security.nginx.redirect_to_https")
-@post_change_hook("security.nginx.compatibility")
-@post_change_hook("security.webadmin.allowlist.enabled")
-@post_change_hook("security.webadmin.allowlist")
+@post_change_hook("ssowat_panel_overlay_enabled")
+@post_change_hook("nginx_redirect_to_https")
+@post_change_hook("nginx_compatibility")
+@post_change_hook("webadmin_allowlist_enabled")
+@post_change_hook("webadmin_allowlist")
 def reconfigure_nginx(setting_name, old_value, new_value):
     if old_value != new_value:
         regen_conf(names=["nginx"])
 
 
-@post_change_hook("security.experimental.enabled")
+@post_change_hook("security_experimental_enabled")
 def reconfigure_nginx_and_yunohost(setting_name, old_value, new_value):
     if old_value != new_value:
         regen_conf(names=["nginx", "yunohost"])
 
 
-@post_change_hook("security.ssh.compatibility")
-@post_change_hook("security.ssh.password_authentication")
+@post_change_hook("ssh_compatibility")
+@post_change_hook("ssh_password_authentication")
 def reconfigure_ssh(setting_name, old_value, new_value):
     if old_value != new_value:
         regen_conf(names=["ssh"])
 
 
-@post_change_hook("security.ssh.port")
+@post_change_hook("ssh_port")
 def reconfigure_ssh_and_fail2ban(setting_name, old_value, new_value):
     if old_value != new_value:
         regen_conf(names=["ssh", "fail2ban"])
         firewall_reload()
 
 
-@post_change_hook("smtp.allow_ipv6")
-@post_change_hook("smtp.relay.host")
-@post_change_hook("smtp.relay.port")
-@post_change_hook("smtp.relay.user")
-@post_change_hook("smtp.relay.password")
-@post_change_hook("security.postfix.compatibility")
+@post_change_hook("smtp_allow_ipv6")
+@post_change_hook("smtp_relay_host")
+@post_change_hook("smtp_relay_port")
+@post_change_hook("smtp_relay_user")
+@post_change_hook("smtp_relay_password")
+@post_change_hook("postfix_compatibility")
 def reconfigure_postfix(setting_name, old_value, new_value):
     if old_value != new_value:
         regen_conf(names=["postfix"])
```

Comment:
It feels like it would be more elegant to have some way to tell directly in the settings' toml that we want to regen-conf some stuff when these gets changed ... but that wouldnt cover more complex cases such as `pop3_enabled` where we trigger and entire `apt install` etc ... maybe that's no big deal and we can keep this as is for now